Build Error: Minecraft 1.21.11

## Build Error Report

**Minecraft Version:** 1.21.11
**Commit:** 9253d3b
**Error:** ShellError: Failed with exit code 1

### Full Build Logs

```
ShellError: Failed with exit code 1

To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/9.2.1/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
Daemon will be stopped at the end of the build 

> Configure project :
Fabric Loom: 1.14.10
Cannot remap field_4165 because it does not exists in any of the targets [net/minecraft/class_775] or their parents.
Cannot remap field_4166 because it does not exists in any of the targets [net/minecraft/class_775] or their parents.
Cannot remap method_27640 because it does not exists in any of the targets [net/minecraft/class_5235$class_5241$1] or their parents.
Cannot remap method_73757 because it does not exists in any of the targets [net/minecraft/class_11993] or their parents.
Cannot remap field_19419 because it does not exists in any of the targets [net/minecraft/class_3065$1] or their parents.

Cannot remap method_8359 because it does not exists in any of the targets [net/minecraft/class_1928] or their parents.
Cannot remap field_9197 because it does not exists in any of the targets [net/minecraft/class_1928] or their parents.
Cannot remap method_20762 because it does not exists in any of the targets [net/minecraft/class_3065$1]
Cannot remap <init> because it does not exists in any of the targets [net/minecraft/class_5350]
Cannot remap method_73754 because it does not exists in any of the targets [net/minecraft/class_11993]
Cannot remap method_73757 because it does not exists in any of the targets [net/minecraft/class_11993]
Cannot remap method_7246 because it does not exists in any of the targets [net/minecraft/class_3853$class_4162]
Cannot remap method_27640 because it does not exists in any of the targets [net/minecraft/class_5235$class_5241$1]
Cannot remap field_24609 because it does not exists in any of the targets [] or their parents.
Cannot remap method_2468 because it does not exists in any of the targets [net/minecraft/class_481]
Cannot remap method_58296 because it does not exists in any of the targets [net/minecraft/class_5350]
Cannot remap method_13394 because it does not exists in any of the targets [net/minecraft/class_3065] or their parents.
Cannot remap method_13397 because it does not exists in any of the targets [net/minecraft/class_3065] or their parents.
Cannot remap method_25423 because it does not exists in any of the targets [net/minecraft/class_437]
Cannot remap method_25423 because it does not exists in any of the targets [net/minecraft/class_437]
Cannot remap method_25410 because it does not exists in any of the targets [net/minecraft/class_437]
Cannot remap method_25410 because it does not exists in any of the targets [net/minecraft/class_437]
Cannot remap method_18097 because it does not exists in any of the targets [net/minecraft/class_310]
Cannot remap method_3345 because it does not exists in any of the targets [net/minecraft/class_775]
Cannot remap method_18096 because it does not exists in any of the targets [net/minecraft/class_310]
Cannot remap method_18096 because it does not exists in any of the targets [net/minecraft/class_310]
Cannot remap method_18096 because it does not exists in any of the targets [net/minecraft/class_310]
Cannot remap method_20760 because it does not exists in any of the targets [] or their parents.
Cannot remap <init> because it does not exists in any of the targets [net/minecraft/class_5350]
Cannot remap method_18096 because it does not exists in any of the targets [net/minecraft/class_310]
Cannot remap field_55271 because it does not exists in any of the targets [net/minecraft/class_776] or their parents.
Cannot remap method_29466 because it does not exists in any of the targets [net/minecraft/class_5350]

> Configure project :
:remapping sources

> Task :compileJava
/tmp/mod-z1zuncrj1qn/src/main/java/me/imgalvin/autotorch/Autotorch.java:88: error: cannot find symbol
                        .then(literal("debug").requires(source -> source.hasPermissionLevel(2)).executes(context -> {
                                                                        ^
  symbol:   method hasPermissionLevel(int)
  location: variable source of type ServerCommandSource
1 error

> Task :compileJava FAILED

[Incubating] Problems report is available at: file:///tmp/mod-z1zuncrj1qn/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler output below.
  /tmp/mod-z1zuncrj1qn/src/main/java/me/imgalvin/autotorch/Autotorch.java:88: error: cannot find symbol
                          .then(literal("debug").requires(source -> source.hasPermissionLevel(2)).executes(context -> {
                                                                          ^
    symbol:   method hasPermissionLevel(int)
    location: variable source of type ServerCommandSource
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (powered by Develocity).

BUILD FAILED in 1m 1s
1 actionable task: 1 executed

```

---
*This issue was automatically created by the Minecraft CI system.*

--- a/README.md
+++ b/README.md
@@ -16,11 +16,19 @@ _A torch a day keeps the mobs away!_ Or something like that.
 Just keep torches in your inventory, and you'll find that the surrounding space will be automatically lit up!
 
 # Supported versions
+[![Supported versions](https://modrinth-shields.imgalvin.me/badge/autotorcher)](https://modrinth.com/mod/autotorcher)
+
 AutoTorcher is available for Minecraft versions 1.21.10 and above!
 
-Due to the frequency of Minecraft updates now, each Minecraft version has its own .jar file. This is mainly to prevent crashes from each minor update
+Due to the frequency of Minecraft updates now, each Minecraft version has its own .jar file. This is mainly to prevent crashes from each drop.
+
+Please also keep your loader and Fabric API up to date to ensure the best compatibility!
 
 # Changelog
+## 1.2.0 - 30/12/2025
+- Updated for Minecraft 1.21.11
+- Removes torches from being placed when in adventure mode
+
 ## 1.1.0 - 14/12/2025
 Closes [#2](https://github.com/GalvinCraft/auto-torch-fabric/issues/2)
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -7,15 +7,15 @@ org.gradle.configuration-cache=false
 
 # Fabric Properties
 # check these on https://fabricmc.net/develop
-minecraft_version=1.21.10
-yarn_mappings=1.21.10+build.2
+minecraft_version=1.21.11
+yarn_mappings=1.21.11+build.3
 loader_version=0.17.3
-loom_version=1.11-SNAPSHOT
+loom_version=1.14-SNAPSHOT
 
 # Mod Properties
-mod_version=1.1.0+1.21.10
+mod_version=1.2.0+1.21.11
 maven_group=me.imgalvin.autotorch
 archives_base_name=auto-torch
 
 # Dependencies
-fabric_version=0.136.0+1.21.10
+fabric_version=0.140.2+1.21.11

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-9.1.0-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-9.2.1-bin.zip
 networkTimeout=10000
 validateDistributionUrl=true
 zipStoreBase=GRADLE_USER_HOME

--- a/src/main/java/me/imgalvin/autotorch/Autotorch.java
+++ b/src/main/java/me/imgalvin/autotorch/Autotorch.java
@@ -12,6 +12,7 @@ import net.fabricmc.fabric.api.networking.v1.ServerPlayConnectionEvents;
 import net.minecraft.block.Blocks;
 import net.minecraft.entity.player.PlayerInventory;
 import net.minecraft.item.Items;
+import net.minecraft.server.command.CommandManager;
 import net.minecraft.server.command.ServerCommandSource;
 import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.text.Text;
@@ -85,7 +86,7 @@ public class Autotorch implements ModInitializer {
                         .then(literal("on").executes(context -> setAutoTorchState(context, true)))
                         .then(literal("off").executes(context -> setAutoTorchState(context, false)))
                         .then(literal("toggle").executes(Autotorch::toggleAutoTorchState))
-                        .then(literal("debug").requires(source -> source.hasPermissionLevel(2)).executes(context -> {
+                        .then(literal("debug").requires(CommandManager.requirePermissionLevel(CommandManager.GAMEMASTERS_CHECK)).executes(context -> {
                             debugLoggingEnabled = !debugLoggingEnabled;
                             String status = debugLoggingEnabled ? "enabled" : "disabled";
                             Formatting color = debugLoggingEnabled ? Formatting.GREEN : Formatting.YELLOW;
@@ -121,8 +122,10 @@ public class Autotorch implements ModInitializer {
                         continue;
                     }
 
-                    // Also only do this if the player is in survival or adventure mode
-                    if (!player.interactionManager.getGameMode().isSurvivalLike()) {
+                    // Also only do this if the player is in survival mode
+                    // isBlockBreakingRestricted() returns true for adventure and spectator modes
+                    // isCreative() returns true for creative mode
+                    if (player.interactionManager.getGameMode().isBlockBreakingRestricted() || player.isCreative()) {
                         continue;
                     }
 

--- a/src/main/java/me/imgalvin/autotorch/Autotorch.java
+++ b/src/main/java/me/imgalvin/autotorch/Autotorch.java
@@ -122,9 +122,9 @@ public class Autotorch implements ModInitializer {
                         continue;
                     }
 
-                    // Also only do this if the player is in survival mode
-                    // isBlockBreakingRestricted() returns true for adventure and spectator modes
-                    // isCreative() returns true for creative mode
+                    // Only run this logic if the player is in survival mode (not adventure, spectator, or creative).
+                    // isBlockBreakingRestricted() returns true for adventure and spectator modes, which we want to skip.
+                    // isCreative() returns true for creative mode, which we also want to skip.
                     if (player.interactionManager.getGameMode().isBlockBreakingRestricted() || player.isCreative()) {
                         continue;
                     }

--- a/src/main/resources/fabric.mod.json
+++ b/src/main/resources/fabric.mod.json
@@ -22,8 +22,8 @@
 	},
 	"depends": {
       "fabricloader": ">=0.17.3",
-      "minecraft": "1.21.10",
+      "minecraft": "1.21.11",
       "java": ">=21",
-      "fabric-api": "*"
+      "fabric-api": ">=0.139.5"
     }
 }

